patches/chrome-android-java-src-org-chromium-chrome-browser-toolbar-top-ToolbarPhone.java.patch

```diff
@@ -0,0 +1,52 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
+index 25c687987c90ac32f40698e9270cddc1d77f3bb0..0ea457c1b335d38d402d96ae559e792170854c57 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
+@@ -89,7 +89,7 @@ import java.util.List;
+ /**
+  * Phone specific toolbar implementation.
+  */
+-public class ToolbarPhone extends ToolbarLayout implements Invalidator.Client, OnClickListener,
++public class ToolbarPhone extends BraveToolbarLayout implements Invalidator.Client, OnClickListener,
```

Comment:
this can be done with asm

Reply:
It requires code reorganization as other patches here rely on BraveToolbarLayout as a super class:
```
../../chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java:417: error: cannot find symbol
        super.updateModernLocationBarColor(color);                                                        
             ^                                                                                            
  symbol: method updateModernLocationBarColor(int)                                                        
../../chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java:534: error: cannot find symbol
        super.onClick(v);                                                                                 
             ^                                                                                            
  symbol: method onClick(View)                                                                            
../../chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java:703: error: cannot find symbol
        return Math.max(mToolbarSidePadding, super.getBoundsAfterAccountingForRightButtons(mToolbarButtonsContainer));
                                                  ^                                                       
  symbol: method getBoundsAfterAccountingForRightButtons(ViewGroup)                                       
../../chrome/android/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java:2137: error: cannot find symbol
        populateUrlAnimatorSet(hasFocus, URL_FOCUS_TOOLBAR_BUTTONS_DURATION_MS, URL_CLEAR_FOCUS_TABSTACK_DELAY_MS, URL_FOCUS_TOOLBAR_BUTTONS_TRANSLATION_X_DP, animators);
        ^                                                                                                 
  symbol:   method populateUrlAnimatorSet(boolean,int,int,int,List<Animator>)                             
  location: class ToolbarPhone                                                                            
../../brave/android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:243: error: incompatible types: BraveToolbarLayout cannot be converted to ToolbarPhone
    if (this instanceof ToolbarPhone) {                                                                   
        ^                                                                                                 
../../brave/android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:884: error: incompatible types: BraveToolbarLayout cannot be converted to ToolbarPhone
      if (this instanceof ToolbarPhone && super.getMenuButtonCoordinator() != null) {                     
          ^                                                                                               
7 errors
```

Reply:
This requires quite substantial changes that need to be re-tested, besides these changes are not new in cr87 and came from master and should be fixed and tested within master. Follow up issue created https://github.com/brave/brave-browser/issues/12322